src/main/java/org/jenkinsci/test/acceptance/po/Login.java

```diff
@@ -64,7 +58,11 @@ public Login doLogin(User user) {
 
     public Login doSuccessfulLogin(String user, String password) {
         this.doLogin(user, password);
-        Assert.assertThat(this, loggedInAs(user));
+        waitFor().withTimeout(30, TimeUnit.SECONDS).until(() -> {
+            assertThat(this, not(hasInvalidLoginInformation())); // login hasn't failed
+            return ExpectedConditions.visibilityOfElementLocated(by.id("jenkins")); // redirect has happened
```

Comment:
I am wondering if this can happen on other places where we also use `clickAndWaitToBecomeStale` or until(CapybaraPortingLayerImpl::isStale);

This could likely be the cause (and the fix) of  the race condition on creating a new `Folder` that has been observed a a few times. https://issues.jenkins.io/browse/JENKINS-64358

Reply:
Yes it looks like the same class of issue.